pkg/controller/issuer/certificate/reconciler.go

```diff
@@ -549,6 +556,66 @@ func (r *certReconciler) restoreCA(issuerKey utils.IssuerKey, issuer *api.Issuer
 	return CAKeyPair, nil
 }
 
+func (r *certReconciler) obtainCertificateSelfSigned(logctx logger.LogContext, obj resources.Object,
+	renew bool, cert *api.Certificate, issuerKey utils.IssuerKey, issuer *api.Issuer) reconcile.Status {
+	if cert.Spec.IsCA == nil || !*cert.Spec.IsCA {
+		return r.failedStop(logctx, obj, api.StateError, fmt.Errorf("Self signed certificates must set 'spec.isCA: true'"))
+	}
+	duration := legobridge.DefaultCertDuration
+	if cert.Spec.Duration != nil {
+		if cert.Spec.Duration.Duration <= r.renewalWindow {
```

Comment:
We should have a larger value for the duration in relation to the renewal window
```suggestion
		if cert.Spec.Duration.Duration < 2 * r.renewalWindow {
```